### PR TITLE
fix: add device arn error handling for badly formed ARNs

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

While device ARN validation is handled by the service, we do some basic preprocessing that assumes the ARN will be of a certain format (ie. having at least 3 :'s). Previously, an ARN that doesn't meet this requirement would cause a crash in the BDK; now this throws an error gracefully.

*Testing done:*

Added a unit test.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/aws/aws_quantum_job.py [hunk 1]
```diff
@@ -528,10 +528,17 @@ def __eq__(self, other) -> bool:
     def __hash__(self) -> int:
         return hash(self.arn)
 
+    @staticmethod
+    def _get_device_region(device_arn: str) -> str:
+        try:
+            return device_arn.split(":")[3]
+        except IndexError:
+            raise ValueError(f"Device arn is not a valid format: {device_arn}")
```
Comment (math411) [inline src/braket/aws/aws_quantum_job.py:536]: Would it make sense to include a global and regional arn as examples in the error output?
```suggestion
            raise ValueError(f"Device arn is not a valid format: {device_arn}"). Example of correct arns are:
            arn:aws:braket:::device/qpu/d-wave/Advantage_system1 or arn:aws:braket:us-west-2::device/qpu/d-wave/Advantage_system1
```

Comment (ajberdy) [reply src/braket/aws/aws_quantum_job.py:536]: Sure, that could make sense. Adding a region to a non-regionalized ARN will not work though, so we'll have to use M1 as an example. Also need to update the test.

File: src/braket/aws/aws_quantum_job.py [hunk 2]
```diff
@@ -528,10 +528,21 @@ def __eq__(self, other) -> bool:
...
 
+    @staticmethod
+    def _get_device_region(device_arn: str) -> str:
+        try:
+            return device_arn.split(":")[3]
+        except IndexError:
+            raise ValueError(
+                f"Device arn is not a valid format: {device_arn}. Examples of correct arns are: "
+                "arn:aws:braket:::device/quantum-simulator/amazon/sv1 or "
+                "arn:aws:braket:us-west-1::device/qpu/rigetti/Aspen-M-1."
```
Comment (kshitijc) [inline src/braket/aws/aws_quantum_job.py:539]: I'm not sure if we should add specific arns here. Might be better to point to the docs specifying the valid arns: https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html

**Review by kshitijc — COMMENTED** [review]
Tagging @aws-rbs for reviewing the error message string.

**Review by kshitijc — COMMENTED** [review]
The same logic exists in aws_device as well. Do we wish to update that as well ?

**Comment by ajberdy** [comment]
> The same logic exists in aws_device as well. Do we wish to update that as well ?

Yes, good call.

**Review by kshitijc — APPROVED** [review]

**Pull request merged**